plone/app/layout/globals/layout.py

```diff
@@ -93,40 +93,27 @@ def have_portlets(self, manager_name, view=None):
 
         return renderer.visible
 
-    @memoize
-    def icons_visible(self):
-        """Returns True if icons should be shown or False otherwise."""
-        context = self.context
-        membership = getToolByName(context, "portal_membership")
-        anon = membership.isAnonymousUser()
-
+    def _image_visibility(self, name):
+        """check if image {name} is visible with current settings and user"""
         registry = getUtility(IRegistry)
         settings = registry.forInterface(ISiteSchema, prefix="plone", check=False)
-        icon_visibility = settings.icon_visibility
-
-        if icon_visibility == "enabled":
-            return True
-        elif icon_visibility == "authenticated" and not anon:
+        visibility = getattr(settings, f"{name}_visibility")
+        if visibility == "enabled":
             return True
-        else:
+        if visibility != "authenticated":
             return False
+        user = getSecurityManager().getUser()
+        return user is not None and user.getUserName() != 'Anonymous User'
```

Comment:
Usually checking for roles or permission is better, but that is fine.

Reply:
Maybe memoizing
```
@memoize
def is_anon(self):
        membership = getToolByName(context, "portal_membership")
        anon = membership.isAnonymousUser()
```
helps?

Reply:
ad1) It is as it was before. I do not want to change behavior.

ad 2) It is faster to use the SecurityManager directly instead of a superfluous expensive `getToolByName` (btw` membership.isAnonymousUser()` does exact this).